MODULE.bazel

```diff
@@ -1,6 +1,49 @@
-###############################################################################
-# Bazel now uses Bzlmod by default to manage external dependencies.
-# Please consider migrating your external dependencies from WORKSPACE to MODULE.bazel.
-#
-# For more details, please check https://github.com/bazelbuild/bazel/issues/18958
-###############################################################################
+module(
+    name = "grpc_ecosystem_grpc_gateway",
+    version = "2.23.0",
+)
+
+# Bazel Central Registry modules.
+bazel_dep(name = "bazel_features", version = "1.20.0")
+bazel_dep(name = "rules_license", version = "1.0.0")
+bazel_dep(name = "rules_python", version = "0.37.2")
+bazel_dep(name = "rules_proto", version = "6.0.2")
+bazel_dep(name = "rules_go", version = "0.50.1", repo_name = "io_bazel_rules_go")
+bazel_dep(name = "gazelle", version = "0.39.1", repo_name = "bazel_gazelle")
+bazel_dep(name = "bazel_skylib", version = "1.7.1")
+bazel_dep(name = "protobuf", version = "26.0.bcr.2", repo_name = "com_google_protobuf")
+bazel_dep(name = "googleapis", version = "0.0.0-20240819-fe8ba054a")
+
+go_sdk = use_extension("@io_bazel_rules_go//go:extensions.bzl", "go_sdk")
+go_sdk.download(version = "1.22.7")
+
+go_deps = use_extension("@bazel_gazelle//:extensions.bzl", "go_deps")
+go_deps.from_file(go_mod = ":go.mod")
+go_deps.module(
+    path = "google.golang.org/grpc/cmd/protoc-gen-go-grpc",
+    sum = "h1:rNBFJjBCOgVr9pWD7rs/knKL4FRTKgpZmsRfV214zcA=",
+    version = "v1.3.0",
+)
+go_deps.module(
+    path = "github.com/golang/protobuf",
+    sum = "h1:i7eJL8qZTpSEXOPTxNKhASYpMn+8e5Q6AdndVa1dWek=",
+    version = "v1.5.4",
+)
```

Comment:
These are necessary because Gazelle only generates dependencies for Go files (since it generates dependencies by reading the `go.mod` file).

These two dependencies are introduced by `.proto` files and are not included in `go.mod`, so by adding these we tell Gazelle that those should be downloaded as well.

Reply:
I'm confused. If they are go deps, they should be in the go.mod file, since we check in our protobuf files and any dependencies they have will form part of the go.mod dependency closure at Go dependency resolution time. If these are pure protobuf dependencies (AFAIK, we don't have any proto dependencies other than googleapis), why are they part of `go_deps`?

Reply:
Great question.

Basically, both of these repositories were used in targets that are not managed by Gazelle.

1. `google.golang.org/grpc/cmd/protoc-gen-go-grpc` is used in a [`go_proto_compiler` target](https://github.com/grpc-ecosystem/grpc-gateway/blob/86a430974ccf379524c80a4cd89759c8f5b91d4c/BUILD.bazel#L62). Gazelle does not manage `go_proto_compiler` targets.
2. `github.com/golang/protobuf` is used in a [`go_proto_library` target that compiles example protos to go](https://github.com/grpc-ecosystem/grpc-gateway/blob/86a430974ccf379524c80a4cd89759c8f5b91d4c/examples/internal/proto/examplepb/BUILD.bazel#L103-L124). While `go_proto_library` is managed by Gazelle, there are a number of [Gazelle directives at the beginning of this file](https://github.com/grpc-ecosystem/grpc-gateway/blob/86a430974ccf379524c80a4cd89759c8f5b91d4c/examples/internal/proto/examplepb/BUILD.bazel#L10-L37) that tell it to not manage the Go files compiled from these protos, which causes `go.mod` to not include the dependency.

Note that this was not an issue in the `WORKSPACE` version because we had a giant file that defined both dependencies manually ([1](https://github.com/grpc-ecosystem/grpc-gateway/blob/86a430974ccf379524c80a4cd89759c8f5b91d4c/repositories.bzl#L306-L311), [2](https://github.com/grpc-ecosystem/grpc-gateway/blob/86a430974ccf379524c80a4cd89759c8f5b91d4c/repositories.bzl#L1819-L1824)).

Unless we change the excludes, these dependencies will have to be specified manually.

Reply:
Thanks for the explanation, I'll categorize this in my mind as "because bazel".

Reply:
Haha, yeah, pretty much.